spec/models/service_template_ansible_playbook_spec.rb

```diff
@@ -237,14 +237,43 @@
 
       expect(service_template.dialogs.first.id).to eq info[:service_dialog_id]
     end
+  end
+
+  describe '#destroy' do
+    it 'destroys a job template if there is a playbook_id passed in' do
+      service_template = prebuild_service_template
 
-    def prebuild_service_template
+      expect(ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook).to receive(:find).and_return(playbook).twice
+      expect(ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript)
+        .to receive(:delete_in_provider_queue).twice
+      service_template.destroy
+    end
+
+    it 'does not destroy a job template if no playbook_id is passed in' do
       expect(described_class)
         .to receive(:create_job_templates).and_return(:provision => {:configuration_template => job_template})
+        .at_least(:once)
+
       service_template = described_class.create_catalog_item(catalog_item_options_two, user)
-      expect(service_template).to receive(:job_template)
-        .and_return(job_template).at_least(:once)
-      service_template
+
+      expect(ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook).to receive(:find).and_return(playbook).never
+      expect(ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript)
+        .to receive(:delete_in_provider_queue).never
+      [:provision, :retirement].each do |action|
+        service_template.options[:config_info][action].delete(:playbook_id)
+      end
+      service_template.destroy
     end
   end
+
+  def prebuild_service_template
```

Comment:
This could be:
```ruby
def prebuild_service_template
  return = {:provision => {:configuration_template => job_template}}
  expect(described_class).to receive(:create_job_templates).and_return(return).at_least(:once)
  described_class.create_catalog_item(catalog_item_options_two, user).tap do |service_template|
    expect(service_template).to receive(:job_template).and_return(job_template).at_least(:once)
  end
end
```